nodebuilder/tests/sync_test.go

```diff
@@ -265,54 +256,52 @@ Steps:
 3. Check BN is synced to height 20
 4. Create a Full Node(FN) with a connection to BN as a trusted peer
 5. Start a FN
-6. Check FN is synced to height 30
-7. Create a Light Node(LN) with a connection to BN, FN as trusted peers
+6. Check FN is synced to network head
+7. Create a Light Node(LN) with a connection to BN and FN as trusted peers
 8. Start LN
-9. Check LN is synced to height 50
+9. Check LN is synced to network head.
 */
 func TestSyncLightWithTrustedPeers(t *testing.T) {
-	sw := swamp.NewSwamp(t)
-
-	bridge := sw.NewBridgeNode()
-
 	ctx, cancel := context.WithTimeout(context.Background(), swamp.DefaultTestTimeout)
 	t.Cleanup(cancel)
 
-	sw.WaitTillHeight(ctx, 20)
+	sw := swamp.NewSwamp(t)
+	fillDn := swamp.FillBlocks(ctx, sw.ClientContext, sw.Accounts, bsize, numBlocks)
+	sw.WaitTillHeight(ctx, numBlocks)
 
+	// create a BN and let it sync to network head
+	bridge := sw.NewBridgeNode()
 	err := bridge.Start(ctx)
 	require.NoError(t, err)
-
-	h, err := bridge.HeaderServ.GetByHeight(ctx, 20)
-	require.NoError(t, err)
-
-	assert.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, 20))
-
-	addrs, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(bridge.Host))
+	err = bridge.HeaderServ.SyncWait(ctx)
 	require.NoError(t, err)
 
+	// create a FN with BN as trusted peer
 	cfg := nodebuilder.DefaultConfig(node.Full)
-	cfg.Header.TrustedPeers = append(cfg.Header.TrustedPeers, addrs[0].String())
+	bridgeInfo, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(bridge.Host))
+	require.NoError(t, err)
+	cfg.Header.TrustedPeers = append(cfg.Header.TrustedPeers, bridgeInfo[0].String())
 	full := sw.NewNodeWithConfig(node.Full, cfg)
-	require.NoError(t, full.Start(ctx))
 
-	h, err = full.HeaderServ.GetByHeight(ctx, 30)
+	// let FN sync to network head
+	err = full.Start(ctx)
 	require.NoError(t, err)
-
-	assert.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, 30))
-
-	addrs, err = peer.AddrInfoToP2pAddrs(host.InfoFromHost(full.Host))
+	err = full.HeaderServ.SyncWait(ctx)
 	require.NoError(t, err)
 
+	// create a LN with both FN and BN as trusted peers
 	cfg = nodebuilder.DefaultConfig(node.Light)
-	cfg.Header.TrustedPeers = append(cfg.Header.TrustedPeers, addrs[0].String())
+	fullInfo, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(full.Host))
+	require.NoError(t, err)
+	cfg.Header.TrustedPeers = append(cfg.Header.TrustedPeers, bridgeInfo[0].String(), fullInfo[0].String())
 	light := sw.NewNodeWithConfig(node.Light, cfg)
 
+	// let LN sync to network head
 	err = light.Start(ctx)
 	require.NoError(t, err)
-
-	h, err = light.HeaderServ.GetByHeight(ctx, 50)
+	err = light.HeaderServ.SyncWait(ctx)
 	require.NoError(t, err)
 
-	assert.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, 50))
+	// wait for the core block filling process to exit
+	require.NoError(t, <-fillDn)
```

Comment:
we don't trust that chans will get closed

Reply:
Relates to all places

Reply:
will wrap w select on ctx

Reply:
https://github.com/celestiaorg/celestia-node/pull/2162/commits/0c19eb09514b56d1dad13d4ac251270cf9c2f2a1